lib/hello_sign/resource/base_resource.rb

```diff
@@ -40,8 +42,9 @@ class BaseResource
       #
       # @return [HelloSign::Resource::BaseResource] a new BaseResource
       def initialize(hash, key=nil)
-        @raw_data = key ? hash[key] : hash
-        @warnings = hash['warnings'] ? hash['warnings'] : nil
+        @headers = hash[:headers]
+        @raw_data = key ? hash[:body][key] : hash
+        @warnings = hash['warnings'] ? hash[:body]['warnings'] : nil
```

Comment:
Shouldn't this be: 
@warnings = hash**[:body]**['warnings'] ? hash[:body]['warnings'] : nil

Reply:
may want to add a test for warnings as well